src/Symfony/Component/Validator/Util/PropertyPath.php

```diff
@@ -42,7 +42,7 @@ public static function append($basePath, $subPath)
                 return $basePath.$subPath;
             }
 
-            return $basePath ? $basePath.'.'.$subPath : $subPath;
+            return '' !== (string) $basePath ? $basePath.'.'.$subPath : $subPath;
```

Comment:
I think you wont need the explicit  string-cast when strictly comparing against `''`